### ISSUE TITLE
Flutter Web - random but consistent and reproducible glitches and crashes on Chromium

### ISSUE THREAD

**Issue by obeobe** [flutter/flutter#88643]
Flutter Web seems quite unstable on Chromium, both when running with the debugger, and when running without it.

### Update (read first):
The below issues DO NOT reproduce for me with Chrome (92.0.4515.159, 64-bit).
They do reproduce for me on Chromium, but its an older version - 86.0.4240.75.
So it's possible these issues have already been fixed, but unfortunately I can't currently update my Chromium to the latest version to verify.

I also discovered something else that may be relevant.
I discovered that after closing Chromium and ending a debug session - several instances of Chromium are left running (i.e. they can be seen with `ps aux | grep chromium`). If I forcibly `kill -9` them - then the issues I describe below take a much longer time to reproduce (maybe 20-30 minutes).

Considering all of this, maybe it's not worth investigating this issue at the moment.
I'm not sure if I should close or delete it, so I just leave these notes.

### Original report:
I keep seeing behaviors like:
* A "RuntimeError: function signature mismatch" in the Console (very common).
* Chromium "Aw, Snap!" SIGSEGV crashes (common).
* Visual glitches, like visual elements disappearing, or the entire viewport suddenly rotated by 45 degrees (less common).

These issues appear randomly, but consistently. They don't require user interaction with the browser to happen.

I was able to consistently reproduce all three by modifying the bootstrap Flutter application (the counter thing) so that the program increases `_counter` every 100ms.

I did the 100ms timer thing to expedite these problems, but I've seen them happen in my actual application without any kind of extreme conditions. E.g. while typing some text, or when just moving the mouse around the screen, or even without any interaction.

I used the program provided below (at the bottom) to reproduce these issues.
It increases the counter by 1 every 100ms, and logs it to the console every 1s.
I ran the program six times and documented the results:

```
Run #1:
	* "RuntimeError: function signature mismatch" at #360 (i.e. when `_counter` was between 360 and 369).
	* SIGSEGV at #2130

Run #2:
	* "RuntimeError: function signature mismatch" at #30
	* "RuntimeError: function signature mismatch" at #130
	* at some point before #500 the title, text, and FAB disappeared (and never returned)
	* SIGSEGV at #970

Run #3:
	* "RuntimeError: function signature mismatch" at #370
	* SIGSEGV at #370

Run #4:
	* "RuntimeError: function signature mismatch" at #480
	* browser updates froze at #486 (and never resumed)
	* SIGSEGV at #3110

Run #5:
	* Complete freeze at #377 (the Timer callback is no longer being invoked, FAB not responding, etc.)

Run #6:
	* "RuntimeError: function signature mismatch" at #370
	* "RuntimeError: function signature mismatch" at #720
	* "RuntimeError: function signature mismatch" at #4400
	* SIGSEGV at #5380
```

Notes:
1. I completely closed and restarted Chromium between runs. I.e. I **did not** use hot reload.
2. Here is what the most common error - "RuntimeError: function signature mismatch" looks like in my console:
![image](https://user-images.githubusercontent.com/1394039/130323625-e4734ae9-390e-4878-9c81-c29dbb7f98b5.png)

This is the code I used for the above runs. It's the standard Flutter demo app without the comments and with an `initState` function I added to initialize the `Timer`.
```
import 'dart:async';

import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }


  /////////////////////////////////////
  // ADDED BY ME
  /////////////////////////////////////
  @override
  void initState() {
    super.initState();
    Timer.periodic(new Duration(milliseconds: 100), (timer) {
      this._counter++;
      if (this._counter % 10 == 0) {
        print("counter: ${this._counter}");
      }
      setState(() { });
    });
  }
  /////////////////////////////////////
  // END OF ADDED BY ME
  /////////////////////////////////////


  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headline4,
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}
```

More notes:
3. While experimenting, I saw some other errors, like "Expected a value of type 'SkDeletable', but got one of type 'Null'" and "Unexpected null value." (and others), but they seem less common and overall not destructive to the app.
4. I tested with Chromium on Ubuntu 20.04.
5. `flutter doctor`:
```
[✓] Flutter (Channel stable, 2.2.3, on Linux, locale en_US.UTF-8)
    • Flutter version 2.2.3 at /home/user/flutter
    • Framework revision f4abaa0735 (7 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /home/user/Android/Sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /home/user/Downloads/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+0-b944-P17168821)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/linux#android-setup for more details.

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = chromium

[✓] Android Studio (version 4.2)
    • Android Studio at /home/user/Downloads/android-studio
    • Flutter plugin version 58.0.1
    • Dart plugin version 202.8488
    • Java version OpenJDK Runtime Environment (build 11.0.8+0-b944-P17168821)

[✓] IntelliJ IDEA Ultimate Edition (version 2020.2)
    • IntelliJ at /snap/intellij-idea-ultimate/current
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Chromium 86.0.4240.75 snap
```

**Comment by TahaTesser** [comment]
Hi @obeobe 
if the browser update fixes the issue, how it makes a Flutter bug? 

> Chromium "Aw, Snap!" SIGSEGV crashes (common).

I can confirm this was reported in https://github.com/flutter/flutter/issues/83414

Please upgrade your browser, as the browser issues are related to Flutter 

> The below issues DO NOT reproduce for me with Chrome (92.0.4515.159, 64-bit).

As you confirmed, this is not an issue with the newer browser 

Closing, as this isn't an issue with Flutter itself. If you disagree, please write in the comments and I will reopen it.
Thank you

**Comment by obeobe** [comment]
Hi @TahaTesser ,

Like I said at the top - 
1. I didn't know that it doesn't reproduce in Chrome when I opened this issue. I haven't tried (or even installed Chrome at that time).
2. I still don't know for sure if it's because of differences between Chromium or Chrome, or because of the version. But after I found out it was ok in Chrome I wanted to mention it at the top so others won't unnecessarily waste time on this issue.

I agree with closing this issue. Thanks!
